contracts/vesting/src/vesting.rs

```diff
@@ -0,0 +1,559 @@
+        let block_time = block_time.unwrap_or(env.block.time);
+        let period = self.get_current_vesting_period(block_time);
+
+        match period {
+            // We're in the first period, or the vesting has not started yet.
+            0 => Coin {
+                amount: Uint128(0),
+                denom: DENOM.to_string(),
+            },
+            1..=7 => Coin {
+                amount: Uint128(self.tokens_per_period() * period as u128),
+                denom: DENOM.to_string(),
+            },
+            _ => Coin {
+                amount: self.coin.amount,
+                denom: DENOM.to_string(),
+            },
+        }
+    }
+
+    fn get_vesting_coins(&self, block_time: Option<Timestamp>, env: &Env) -> Coin {
+        Coin {
+            amount: Uint128(
+                self.get_original_vesting().amount.u128()
+                    - self.get_vested_coins(block_time, env).amount.u128(),
+            ),
+            denom: DENOM.to_string(),
+        }
+    }
+
+    fn get_start_time(&self) -> Timestamp {
+        self.start_time
+    }
+
+    fn get_end_time(&self) -> Timestamp {
+        self.periods[(NUM_VESTING_PERIODS - 1) as usize].end_time()
+    }
+
+    fn get_original_vesting(&self) -> Coin {
+        self.coin.clone()
+    }
+
+    fn get_delegated_free(
+        &self,
+        block_time: Option<Timestamp>,
+        env: &Env,
+        storage: &dyn Storage,
+    ) -> Coin {
+        self.get_delegated_with_op(&lt, block_time, env, storage)
+    }
+
+    fn get_delegated_vesting(
+        &self,
+        block_time: Option<Timestamp>,
+        env: &Env,
+        storage: &dyn Storage,
+    ) -> Coin {
+        self.get_delegated_with_op(&ge, block_time, env, storage)
+    }
+}
+
+fn lt(x: u64, y: u64) -> bool {
+    x < y
+}
+
+fn ge(x: u64, y: u64) -> bool {
+    x >= y
+}
+
+pub fn populate_vesting_periods(start_time: u64, n: u64) -> Vec<VestingPeriod> {
+    let mut periods = Vec::new();
+    // There are eight 3 month periods in two years
+    for i in 0..n {
+        let period = VestingPeriod {
+            start_time: start_time + i * VESTING_PERIOD,
+        };
+        periods.push(period);
+    }
+    periods.shrink_to_fit();
+    periods
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::contract::{NUM_VESTING_PERIODS, VESTING_PERIOD};
+    use crate::storage::{get_account, get_account_balance, get_account_delegations};
+    use crate::support::tests::helpers::{init_contract, vesting_account_fixture};
+    use crate::vesting::{DelegationData, VestingAccount};
+    use config::defaults::DENOM;
+    use cosmwasm_std::testing::{mock_dependencies, mock_env, mock_info};
+    use cosmwasm_std::{Addr, Coin, Timestamp, Uint128};
+
+    use super::DelegationAccount;
+
+    #[test]
+    fn test_account_creation() {
+        let mut deps = init_contract();
+        let env = mock_env();
+        let account = vesting_account_fixture(&mut deps.storage, &env);
```

Comment:
This test function kinda feels as it should be testing the actual constructor for the vesting account rather than using the fixture

Reply:
Yup, I just figured that out, as I broke the constructor msg, but my tests were still passing :)